File: backends/gtk4/src/layout/scroll.rs
//! Scroll view implementation for GTK4.

use gtk4::{PolicyType, ScrolledWindow, Widget, prelude::*};
use waterui::{
    Environment, View,
    component::layout::scroll::{Axis, ScrollView},
};

use crate::renderer::render;

/// Render a scroll view using GTK4 ScrolledWindow.
pub fn render_scroll_view<V: View>(scroll: ScrollView, env: &Environment) -> Widget {
    let scrolled_window = ScrolledWindow::new();

    // Set scroll policies
    let h_policy = match scroll.axis {
        Axis::Horizontal => PolicyType::Automatic,
        Axis::Vertical => PolicyType::Never,
        Axis::All => PolicyType::Automatic,
    };

    let v_policy = match scroll.axis {
        Axis::Horizontal => PolicyType::Never,
        Axis::Vertical => PolicyType::Automatic,
        Axis::All => PolicyType::Automatic,
    };

    scrolled_window.set_policy(h_policy, v_policy);

    // Add content
    let content_widget = render(scroll.content, env);
    scrolled_window.set_child(Some(&content_widget));

    scrolled_window.upcast()
}

File: backends/swift/Sources/WaterUI/Scroll.swift
//
//  Scroll.swift
//  waterui-swift
//
//  Created by Lexo Liu on 10/20/24.
//
import SwiftUI
import CWaterUI
/*
struct ScrollView:View,Component{
    static var id=waterui_scroll_id()
    var axis:SwiftUI.Axis.Set
    var content:WaterUI.AnyView
    var body: some View{
        SwiftUI.ScrollView(axis, content: {
            content
        })
    }
    
    init(anyview: OpaquePointer, env: Environment) {
        self.init(scroll: waterui_force_as_scroll(anyview), env: env)
    }
    
    init(scroll: WuiScroll, env: Environment) {
        switch scroll.axis{
        case WATERUI_AXIS_VERTICAL:
            self.axis=[.vertical]
        case WATERUI_AXIS_HORIZONTAL:
            self.axis=[.horizontal]
            
        default:
            self.axis=[.vertical,.horizontal]
        }
        
        self.content=AnyView(anyview: scroll.content, env: env)
    }
}
*/
